salt/states/firewalld.py

```diff
@@ -78,7 +80,60 @@ def present(name,
             masquerade=False,
             ports=None,
             port_fwd=None,
-            services=None):
+            services=None,
+            interfaces=None,
+            sources=None,
+            rich_rules=None,
+            runtime=True,
+            persist=True):
+
+    '''
+    Ensure a zone has specific attributes.
+    '''
+
+    ret = {'name': name,
+           'result': False,
+           'changes': {},
+           'comment': ''}
+
+    if runtime:
+        ret_runtime = _present(name, False, block_icmp, default, masquerade, ports,
+                               port_fwd, services, interfaces, sources, rich_rules)
+
+        for key, value in ret_runtime['changes'].items():
+            ret['changes'].update({key: value})
+
+    if persist:
+        ret_persist = _present(name, True, block_icmp, default, masquerade, ports,
+                               port_fwd, services, interfaces, sources, rich_rules)
+
+        for key, value in ret_persist['changes'].items():
+            ret['changes'].update({key + "_permanent": value})
```

Comment:
It's good if you know python. I'm not mastering python but I can still understand.